### PR TITLE
Can't call HalState.action() without triggering an error

### PR DESCRIPTION
**Pull request by reda-alaoui**

### CODE REVIEW

File: src/state/hal.ts [hunk 1]
```diff
@@ -78,7 +78,7 @@ export class HalState<T = any> extends BaseState<T> {
    */
   action<TFormData = any>(name?: string): Action<TFormData> {
 
-    if (!this.halForm || name !== 'action' || name !== undefined) {
+    if (!this.halForm || name !== undefined) {
```
Comment (evert) [inline src/state/hal.ts:81]: This was obviously not tested well yet, apologies :/

Comment (reda-alaoui) [reply src/state/hal.ts:81]: Don't expect me to complain. 
This is just the best hypermedia client library ever :D

Comment (evert) [reply src/state/hal.ts:81]: Thank you very much! If your usage of Ketting results in something public, I would love to see it!

Comment (reda-alaoui) [reply src/state/hal.ts:81]: For now, I only use it in a protected webapp UI in a B2B context.
If I build something public, I will let you know :)

Comment (evert) [reply src/state/hal.ts:81]: Sweet. Keep in mind that the `actions` part of this lib is still a little experimental, and might undergo a few more changes before it's solid. We're going really start using this ourselves soon (although probably not hal-forms, it's a bit limited) so if I hit walls, I might need to alter how it works.

I'd also be really curious how well it works for you. I don't know if for example returning a `State` object is actually the most logical. So any feedback you have is useful!

Comment (reda-alaoui) [reply src/state/hal.ts:81]: HAL Forms is also experimental from our side. We currently have a single endpoint exposing them.
So the experimental status is ok for us.

I'll let you know how this evolve and contribute if I can do something to help :+1:

**Comment by reda-alaoui** [comment]
Are you going to release this one soon?
Just to know if I release my own fork or wait :)

**Pull request merged**